### PR TITLE
Bump k8s version to the latest 1.22 patch version

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What type of PR is this?**

**What this PR does / why we need it**: Bump k8s versions in tests to v1.22.3.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Bump k8s version to the latest 1.22 patch version
```

### CODE REVIEW

**Comment by CecileRobertMichon** [comment]
> Access to registry 'capzci.azurecr.io' was denied. Response code: 503. Please try running 'az login' again to refresh permissions.

/retest

File: test/e2e/config/azure-dev.yaml [hunk 1]
```diff
@@ -149,7 +149,6 @@ variables:
   EXP_AKS: "true"
   EXP_MACHINE_POOL: "true"
   EXP_CLUSTER_RESOURCE_SET: "true"
-  CONFORMANCE_CI_ARTIFACTS_KUBERNETES_VERSION: "v1.22.1"
```
Comment (CecileRobertMichon) [inline test/e2e/config/azure-dev.yaml:152]: don't think this was used anywhere

**Comment by CecileRobertMichon** [comment]
AKS test is flaky, ref #1800 

/retest

**Comment by CecileRobertMichon** [comment]
The only test that failed is the AKS test which is a known issue, see #1720, so we should be good to merge this

**Review by shysank — COMMENTED** [review]
seeing a bunch of `1.22.1` references in windows patches. Eg. https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/9eafec0ad517461ef36808b5e71d568a1b2ddc12/templates/test/ci/prow-machine-pool-ci-version/patches/kubeadm-bootstrap-windows.yaml#L21. Should we update them too?

**Comment by CecileRobertMichon** [comment]
> seeing a bunch of 1.22.1 references in windows patches. Eg.

verified with @jsturtevant offline

> it doesn't need to be udpated
It's an awful hack because it's actually swapping out the binary under the hood, we just need an image that has that all used
once I can get kube-proxy upstream into k/k we shouldn't need that any more
since they will build ci version for us
and it will be in the k/k build tooling

File: test/e2e/config/azure-dev.yaml [hunk 2]
```diff
@@ -138,7 +138,7 @@ providers:
         new: "--v=2"
 
 variables:
-  KUBERNETES_VERSION: "${KUBERNETES_VERSION:-v1.22.1}"
+  KUBERNETES_VERSION: "${KUBERNETES_VERSION:-v1.22.3}"
```
Comment (CecileRobertMichon) [inline test/e2e/config/azure-dev.yaml:141]: this will actually be overwritten by #1767 when it merges but bumping for now

**Review by CecileRobertMichon — COMMENTED** [review]
/close 

in favor of https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1767/commits/04b2b78317540557156410932cbda69383055f95

**Pull request closed without merging**